src/test/ui/rfc-2093-infer-outlives/check-explicit-dyn.stderr

```diff
@@ -0,0 +1,4 @@
+error: internal compiler error: librustc/ty/subst.rs:493: Type parameter `T/#2` (T/2) out of range when substituting (root type=Some(T)) substs=[ReEarlyBound(0, 'x)]
```

Comment:
@nikomatsakis Not sure how to tackle this error. From some debugging it seems like we are trying to infer `Box: 'x` here. Here is a pastebin link to my debug logs. https://pastebin.com/3fMkHPjb

These are the new debug statements I added: https://github.com/rust-lang/rust/pull/50070/files#diff-a43200d3360f31959b9f923bc33ed1edR251